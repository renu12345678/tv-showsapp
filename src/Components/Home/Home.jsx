import React, { Component } from 'react'
import { Row } from 'antd'
import SearchForm from '../SearchForm/SearchForm'
import Showfilms from '../Showfilms/Showfilms'

export default class Home extends Component {
  constructor (props) {
    super(props)
    this.state = {
      isSearch: false,
      searchValue: '',
      genre:''
    }
  }
  handleSearch = (flag,searchData,genre) => {
    this.setState({ isSearch: flag,searchValue:searchData,genre:genre })
  }

  render () {
    return (
    <div className="container">
    <Row justify="center-around">
      <SearchForm handleSearch={this.handleSearch} />
      { <Showfilms handleSearch={this.handleSearch} {...this.state} genre={this.state.genre}/> }
    </Row>
    </div>
    )
  }
}
