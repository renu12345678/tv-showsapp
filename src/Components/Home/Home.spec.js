import React from 'react';
import { shallow, configure,mount  } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from './Home';


configure({adapter: new Adapter()});
const props = {
    gutter : "0 ",
    justify :"center-around",
    genre: '',
    isSearch: false,
    searchValue: ''
}
let wrapper = mount(<Home {...props}/>);
describe('Home', () => {

  it('should render the Title Component correctly', () => {   
    expect(wrapper.find('div')).not.toBe(null);
  });
  it('should render the Title Component correctly', () => {
    wrapper.instance().handleSearch(false, 'got','drama') 
    expect(wrapper.find('Row')).not.toBe(null);
  });

});