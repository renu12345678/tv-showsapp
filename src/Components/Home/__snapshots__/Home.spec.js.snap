// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the Title Component correctly 1`] = `
<div
  className="container"
>
  <Row
    gutter={0}
    justify="center-around"
  >
    <SearchForm
      handleSearch={[Function]}
    />
    <Showfilms
      genre=""
      handleSearch={[Function]}
      isSearch={false}
      searchValue=""
    />
  </Row>
</div>
`;
